packages/workspace/src/browser/workspace-service.ts

```diff
@@ -80,6 +80,17 @@ export class WorkspaceService implements FrontendApplicationContribution {
             this.server.setRoot(this._root.uri);
         }
     }
+    hasWorkspace: boolean = false;
```

Comment:
Is this really a public mutable property?

Reply:
Created a method hasAWorkspace():boolean, so the field is private